src/Polly.Core/Retry/RetryHelper.cs

```diff
@@ -0,0 +1,31 @@
+using System;
+
+namespace Polly.Retry;
+
+internal static class RetryHelper
+{
+    private const double ExponentialFactor = 2.0;
+
+    public static bool IsValidDelay(TimeSpan delay) => delay >= TimeSpan.Zero;
+
+    public static TimeSpan GetRetryDelay(RetryBackoffType type, int attempt, TimeSpan baseDelay)
+    {
+        if (baseDelay == TimeSpan.Zero)
+        {
+            return baseDelay;
+        }
+
+        return type switch
+        {
+            RetryBackoffType.Constant => baseDelay,
+#if !NETCOREAPP
+            RetryBackoffType.Linear => TimeSpan.FromMilliseconds((attempt + 1) * baseDelay.TotalMilliseconds),
+            RetryBackoffType.Exponential => TimeSpan.FromMilliseconds(Math.Pow(ExponentialFactor, attempt) * baseDelay.TotalMilliseconds),
+#else
+            RetryBackoffType.Linear => (attempt + 1) * baseDelay,
+            RetryBackoffType.Exponential => Math.Pow(ExponentialFactor, attempt) * baseDelay,
+#endif
+            _ => throw new ArgumentOutOfRangeException(nameof(type), type, "The retry backoff type is not supported.")
```

Comment:
In general, I recommend switching to use helpers for throwing exceptions in the code base. Helpers reduce code size and improve i-cache efficiency.

Reply:
I tried it here, but it doesn't work because switch expression expects the TimeSpan result type.

```
error CS0029: Cannot implicitly convert type 'void' to 'System.TimeSpan'
```

I'll leave it as it is for now.